driver-core/src/main/com/mongodb/operation/FindOperation.java

```diff
@@ -1100,6 +1100,7 @@ private CommandCreator getCommandCreator(final SessionContext sessionContext) {
         return new CommandOperationHelper.CommandCreator() {
             @Override
             public BsonDocument create(final ServerDescription serverDescription, final ConnectionDescription connectionDescription) {
+                validateReadConcernAndCollation(connectionDescription, sessionContext.getReadConcern(), collation);
```

Comment:
Not obvious... this is what was causing the FindOperationUnitSpecification test to fail.

Reply:
I plain missed that. Good catch.